### PR TITLE
Use fugit

### PR DESCRIPTION
**Pull request by usbalbin**
This first started with me trying to implement rtc for this crate by stealing it from stm32g0xx-hal. I noticed there were some differences.

This PR changes the time related types to use fugit to make the crate a bit more consistent with stm32g0xx-hal.

Is this at all something we want?

~(Not at all tested on real hardware yet)~ - Some testing done

### CODE REVIEW

File: src/time.rs [hunk 1]
```diff
@@ -1,170 +1,114 @@
-use core::ops::{Add, Div};
+pub use fugit::{
```
Comment (usbalbin) [inline src/time.rs:1]: The contents of this file is now more or less identical with [the one from stm32g0xx-hal](https://github.com/stm32-rs/stm32g0xx-hal/blob/main/src/time.rs)

File: src/timer.rs [hunk 2]
```diff
@@ -87,14 +87,14 @@ impl CountDownTimer<SYST> {
     }
 }
 
-impl CountDown for CountDownTimer<SYST> {
-    type Time = Hertz;
+impl embedded_hal::timer::CountDown for CountDownTimer<SYST> {
+    type Time = MicroSecond;
```
Comment (usbalbin) [inline src/timer.rs:91]: I realize Hertz -> MicroSecond is quite a big change, it felt more natural in my mind. However I am fine with reverting that...

**Comment by usbalbin** [comment]
Any thoughts on this? Is this something that is wanted or should I close this PR?

Also please let me know if you want me to change anything :)

**Review by no111u3 — APPROVED** [review]

**Comment by usbalbin** [comment]
Thanks a lot :) @no111u3

**Pull request merged**